src/inspect_ai/util/_sandbox/local.py

```diff
@@ -43,7 +44,14 @@ async def exec(
         cwd: str | None = None,
         env: dict[str, str] = {},
         timeout: int | None = None,
+        user: str | None = None,
     ) -> ExecResult[str]:
+        if user is not None:
```

Comment:
I think warning here is fine. RE: substitution principle sandboxes aren't at this point intended to fully substitutable, as e.g. a Docker Compose config file can't be read and applied by a K8s environment. 

Side note: while samples/tasks are to _some extent_ intended to work with a given sandbox (especially as they get more complex), it would be nice to converge and author tasks that can be used in a variety of sandboxes. This works today for "default" configurations that don't need a custom Dockerfile or compose config. The Dockerfile could also potentially serve as a lowest common denominator for simpler cases.  Ultimately I am wondering whether we won't need K8s to be able to take a Docker Compose file and synthesise it into a Helm Chart (lossy but some basic variations could probably be made to work).